### DISCUSSION TITLE
Histogram legend shows pre-filtered items

### DISCUSSION THREAD

**Discussion by Chocoder** [biolab/orange3#6027] [Q&A] [answered]
Hello, I love this librairy and I do not know if this issue is a bug or new feature...

My issue: I realized that even after using the raw filtering widget on some datatable, for example on a specific category item, all category appear in the legend of an histogramme widget I add after this very step.

In a table view I see the correct filtering, but as I may have 10 or more items in the initial table it makes the histogramme legend overcharged with useless items...

Is there a trick to show a legend with only existing items?

Maybe the issue exist also on other graphic widgets where I ask a color by category option?

Thanks,

**Comment by janezd** [comment, marked as answer]
Does this help? https://orange3.readthedocs.io/projects/orange-visual-programming/en/latest/widgets/data/purgedomain.html

**Reply by Chocoder** [reply]
hello @janezd ! First thanks for the very fast anwser! I tried this very widget and it works perfectly! I'm newbee in Orange so I guess I'm not aware of this kind of tricks yet...
Moreover do you know if [SHAP](https://shap.readthedocs.io/en/latest/index.html) visuals are "hidden" in some widget or if this kind of visual could be an additionnal feature I could ask for?
Thanks,

**Reply by janezd** [reply]
I'm not familiar with it, but is it anything like the widgets int this add-on? https://orange3-explain.readthedocs.io/en/latest/

**Reply by Chocoder** [reply]
@janezd yes!
Sounds exactly like what I had in mind! I did not see it in the [Widget Catalog](https://orangedatamining.com/widget-catalog/), do you know who could update this very page to offer more visibility to this great work and coding?
Thanks again,

**Reply by ajdapretnar** [reply]
@Chocoder You are right, the widget catalog for Explain somehow doesn't appear in the list. Thanks for pointing this out. We will fix it.

**Reply by Chocoder** [reply]
Happy to contribute with my beginner level 😉
